garageScript/c0d3-app
# fix: Handle flagging an already flagged exercise

## Problem
The current implementation of flagging exercises in the DOJO design doc allows users to flag an exercise multiple times, resulting in a poor user experience. When a user attempts to flag an already flagged exercise, the backend returns an error and the frontend displays "Oops, something went wrong" without providing a clear reason for the error.

## Changes
- Display a message "We appreciate your concern. The exercise has already been reported and we are investigating the problem that has been reported." when a user tries to flag an already flagged exercise.

## Screenshots

<img width="1059" alt="image" src="https://user-images.githubusercontent.com/35906419/212384172-c7280228-ae38-407e-882a-418ef8ac5cd1.png">

## Testing

- Log in to the application
- Navigate to the `/exercises/js0` page
- Attempt to flag an already flagged exercise
- Observe the clear error message displayed

## Concerns

While this change addresses the immediate issue of poor user experience, it does not address the underlying problem of not allowing multiple flags per exercise. It would be better to allow multiple flags per exercise and provide an easy-to-use interface for reviewing and managing exercises with **multiple-flags** in the admin panel.
